storefront/test/helpers/workarea/storefront/schema_org_helper_test.rb

```diff
@@ -0,0 +1,24 @@
+require 'test_helper'
+
+module Workarea
+  module Storefront
+    class SchemaOrgHelperTest < ViewTest
+      include Storefront::Engine.routes.url_helpers
+      include NavigationHelper
+
+      def test_breadcrumb_url_for
+        product = create_product
+        page = create_page
+        url_taxon = create_taxon(url: 'https://www.example.com')
+        navigable_taxon = create_taxon(navigable: page)
+        inventory = create_inventory
+
+        assert_equal(product_url(product, host: Workarea.config.host), breadcrumb_url_for(product))
+        assert_equal(page_url(page, host: Workarea.config.host), breadcrumb_url_for(page))
+        assert_equal('https://www.example.com', breadcrumb_url_for(url_taxon))
+        assert_equal(page_url(page, host: Workarea.config.host), breadcrumb_url_for(navigable_taxon))
+        assert_raises(NoMethodError) { breadcrumb_url_for(inventory) }
```

Comment:
I don't know if we should warn people or render a better error here...the NoMethodError produced when you add something to breadcrumbs that isn't routed at the top level doesn't really tell you much about why it happened...

Reply:
I agree, I think we should look at what the data is upstream causing this situation and see if we can change that.